build: render

jjangga0214 opened the pull request:
[README](https://github.com/awskrug/platform-engineering-glossary/tree/build/render) 를 참고해주시면 감사하겠습니다.

jjangga0214 committed:
build(render): templating with glossary

jjangga0214 committed:
docs(readme): render

sudosubin commented on .github/workflows/lint.yaml:
```diff
@@ -12,21 +12,23 @@ jobs:
         uses: actions/setup-node@v3
         with:
-          node-version: ${{ steps.nvmrc.outputs.version }}
+          node-version: ${{ steps.versions.outputs.nvmrc }}
```
`node-version-file`를 사용하는 건 어떤가요?

jjangga0214 replied:
오 감사해요!!!

sudosubin commented on .github/workflows/lint.yaml:
```diff
@@ -12,21 +12,23 @@ jobs:
         uses: pnpm/action-setup@v2
         with:
-          version: 8.12.1
+          version: ${{ steps.versions.outputs.pnpm }}
```
`package.json`에 `packageManager` field가 있으면 `optional`이라고 하는데 빼는 것은 혹시 어떤가요?

https://github.com/pnpm/action-setup?tab=readme-ov-file#version

jjangga0214 replied:
역시 감사합니다 👍 
기능이 추가된지도 모르고 있었는데 덕분에 전부터 불편하던 부분 2개가 해결되었네요!

jjangga0214 committed:
ci(lint): refactor

jjangga0214 committed:
chore(vscode): rm files.autoSaveDelay

Let teammates use their own config, and patch if and only if they request.

jjangga0214 committed:
docs(readme): fix typo

jjangga0214 committed:
docs(readme): fix typo (2nd)

Geun-Oh commented on test/render.test.ts:
```diff
@@ -0,0 +1,15 @@
+    expect(
+      parseGlossary(`en,ko
+platform engineering,플랫폼 엔지니어링
+IDP,내부 개발자 포탈
```
추가해주신 기능에 관한 사항은 아니지만, 축약어의 경우에는 전체 단어를 기입하도록 하면 좋을 것 같습니다.
https://romaricphilogene.substack.com/p/platform-engineering-7-internal-developer
본래 번역하고자 했던 글인데, 현재는 두 단어 모두 IDP로 통용되고 있어서 헷갈릴 수도 있을 것 같네요 :smiling_face_with_tear:
아니면 위 번역 시에 제가 함께 추가해두겠습니다 :)

jjangga0214 replied:
> 나중에 husky 설정이나 pr comment bot 등등으로 bot이 render해서 생성한 파일과 번역 글이 일치한지 검증도 추가해볼 수 있겠네요. 감사합니다 :)

오 아이디어 공유 감사해요~ 
여담으로 제가 이제껏 개인적으로 하던 생각은 다음과 같았어요 ㅎㅎ

- glossary를 변경하는 PR 이 오면 봇으로 "**1) 영향 받는 아티클 목록**"과 각 "**2) 아티클 내부의 영향받는 line**" 들을 PR 코멘트로 간략히 보여주고, "**3) 각 아티클 작성자(author)들**"을 reviewer 로 지정하는 방안
- `/articles/**/*.md` 또는 glossary 를 변경하는 PR 이 merge 되면 자동으로 dev.to 에 퍼블리시

> 축약어의 경우에는 전체 단어를 기입하도록 하면 좋을 것 같습니다.

아하 그렇네요 이 경우는 둘 다 IDP 네요!!
완벽하게 단일 용어라는 확신이 있는 것 아니면 전체 단어를 기입하는게 좋을 수 있겠네요!

> 아니면 위 번역 시에 제가 함께 추가해두겠습니다 :)

👍

Geun-Oh reviewed (commented):
나중에 husky 설정이나 pr comment bot 등등으로 bot이 render해서 생성한 파일과 번역 글이 일치한지 검증도 추가해볼 수 있겠네요. 감사합니다 :)

jjangga0214 committed:
build(render): refactor variable

jjangga0214 committed:
ci(codeowners): fix syntax

jjangga0214 committed:
ci(lint): fix on(trigger event)

jjangga0214 committed:
revert: "ci(lint): fix on(trigger event)"

This reverts commit 4096516ec734bd02bc71456ef170dae18187aaa7.

jjangga0214 committed:
build(render): fix by upath

dirname(import.meta) returns with \ on Windows,
while globby, which uses fast-glob under the hood,
does not treat as path delimiter.
Thus upath is introduced to solve this issue.

REF: https://github.com/sindresorhus/globby/issues/130
REF: https://github.com/mrmlnc/fast-glob#how-to-write-patterns-on-windows

jjangga0214 committed:
ci: publish and comment
